api/tests/Feature/EventType/EventTypeApiTest.php

```diff
@@ -0,0 +1,266 @@
+        ]);
+
+        $response = $this
+            ->actingAs($user)
+            ->json('GET', self::API_URL . '/' . $eventType->id);
+
+        $response
+            ->assertStatus(JsonResponse::HTTP_OK)
+            ->assertJsonStructure(['data' => self::STRUCTURE]);
+    }
+
+    public function test_get_event_type_by_id_not_found()
+    {
+        $user = factory(User::class)->create([
+            'timezone' => 'Europe/Kiev'
+        ]);
+
+        $this
+            ->actingAs($user)
+            ->json('GET', self::API_URL . '/' . 0)
+            ->assertNotFound();
+    }
+
+    public function test_add_event_type()
+    {
+        $user = factory(User::class)->create([
+            'timezone' => 'Europe/Kiev'
+        ]);
+
+        $response = $this
+            ->actingAs($user)
+            ->json('POST', self::API_URL, self::DATA);
+
+        $response
+            ->assertStatus(JsonResponse::HTTP_CREATED)
+            ->assertJson(['data' => self::DATA])
+            ->assertJson([
+                'data' => [
+                    'owner' => [
+                        'id' => $user->id,
+                        'email' => $user->email,
+                        'name' => $user->name,
+                        'timezone' => $user->timezone,
+                    ]
+                ]
+            ])
+            ->assertJsonStructure(['data' => self::STRUCTURE]);
+    }
+
+    public function test_add_event_type_invalid_request_params()
+    {
+        $user = factory(User::class)->create([
+            'timezone' => 'Europe/Kiev'
+        ]);
+
+        $response = $this
+            ->actingAs($user)
+            ->json('POST', self::API_URL, []);
+
+        $response
+            ->assertStatus(422)
+            ->assertJsonStructure([
+                'error' => [
+                    'message',
+                    'validator',
+                ]
+            ]);
+    }
+
+    public function test_update_event_type_by_id()
+    {
+        $user = factory(User::class)->create([
+            'timezone' => 'Europe/Kiev'
+        ]);
+
+        $eventType = factory(EventType::class)->create([
+            'owner_id' => $user->id,
+        ]);
+
+        $response = $this
+            ->actingAs($user)
+            ->json('PUT', self::API_URL . '/' . $eventType->id, self::DATA);
+
+        $response
+            ->assertStatus(JsonResponse::HTTP_OK)
+            ->assertJson(['data' => self::DATA])
+            ->assertJson([
+                'data' => [
+                    'owner' => [
+                        'id' => $user->id,
+                        'email' => $user->email,
+                        'name' => $user->name,
+                        'timezone' => $user->timezone,
+                    ]
+                ]
+            ])
+            ->assertJsonStructure(['data' => self::STRUCTURE]);
+    }
+
+    public function test_update_event_type_by_id_forbidden()
+    {
+        $users = factory(User::class, 2)->create([
```

Comment:
It's better to create separate variables for users (ex. $authorizedUser, $anotherUser) than working with array indexes. In other places too.